torchvision/transforms/functional.py

```diff
@@ -423,13 +446,32 @@ def resize(
     if not isinstance(interpolation, InterpolationMode):
         raise TypeError("Argument interpolation should be a InterpolationMode")
 
+    if isinstance(size, (list, tuple)):
+        if len(size) not in [1, 2]:
+            raise ValueError(
+                f"Size must be an int or a 1 or 2 element tuple/list, not a {len(size)} element tuple/list"
+            )
+        if max_size is not None and len(size) != 1:
+            raise ValueError(
+                "max_size should only be passed if size specifies the length of the smaller edge, "
+                "i.e. size should be an int or a sequence of length 1 in torchscript mode."
+            )
+
+    _, image_height, image_width = get_dimensions(img)
+    if isinstance(size, int):
+        size = [size]
+    output_size = _compute_output_size((image_height, image_width), size, max_size)
+
+    if (image_height, image_width) == output_size:
+        return img
+
```

Comment:
Sorry, description may not be obvious. The point is the following:
- we refactor duplicated code into a method `_compute_output_size`
- we remove `max_size` arg from private functional modules: `F_t` and `F_pil` where we can do what we want
- For v1 API no breakage as we keep public functional `resize` method without any change in the signature.
- For v2 API, we will implement things in the following way:
  - For image tensor, we reuse `_compute_output_size` and call as previously `F_t.resize`
```python
def resize_image_tensor(
    image: torch.Tensor,
    size: List[int],
    interpolation: InterpolationMode = InterpolationMode.BILINEAR,
    max_size: Optional[int] = None,
    antialias: Optional[bool] = None,
) -> torch.Tensor:
    num_channels, old_height, old_width = get_dimensions_image_tensor(image)
    size = _compute_output_size((old_height, old_width), size=size, max_size=max_size)
    new_height, new_width = size
    batch_shape = image.shape[:-3]
    return _FT.resize(
        image.reshape((-1, num_channels, old_height, old_width)),
        size=size,
        interpolation=interpolation.value,
        antialias=antialias,
    ).reshape(batch_shape + (num_channels, new_height, new_width))
```
  
  - For pillow images, we do the same thing: we reuse `_compute_output_size` and call as previously `F_pil.resize`
  
```python
def resize_image_pil(
    img: PIL.Image.Image,
    size: Union[Sequence[int], int],
    interpolation: InterpolationMode = InterpolationMode.BILINEAR,
    max_size: Optional[int] = None,
) -> PIL.Image.Image:
    if isinstance(size, int):
        size = [size, size]
    # Explicitly cast size to list otherwise mypy issue: incompatible type "Sequence[int]"; expected "List[int]"
    size: List[int] = list(size)
    size = _compute_output_size(img.size[::-1], size=size, max_size=max_size)
    return _FP.resize(img, size, interpolation=pil_modes_mapping[interpolation])
```

  - For bounding boxes we can now implement `max_size` option (which is not the case today):
```python
def resize_bounding_box(
    bounding_box: torch.Tensor, size: List[int], image_size: Tuple[int, int], max_size: Optional[int] = None
) -> torch.Tensor:
    old_height, old_width = image_size
    size = _compute_output_size(image_size, size=size, max_size=max_size)
    new_height, new_width = size
    ratios = torch.tensor((new_width / old_width, new_height / old_height), device=bounding_box.device)
    return bounding_box.view(-1, 2, 2).mul(ratios).view(bounding_box.shape)
```

Source: https://github.com/pytorch/vision/blob/f0c896ff1391dcac098539db79f14e2c50549d7a/torchvision/prototype/transforms/functional/_geometry.py